RadixWallet/Clients/OnLedgerEntitiesClient/OnLedgerEntitiesClient+Interface.swift

```diff
@@ -389,3 +389,92 @@ extension OnLedgerEntitiesClient {
 		}
 	}
 }
+
+extension OnLedgerEntitiesClient {
+	public func isPoolUnitResource(_ resource: OnLedgerEntity.Resource) async throws -> Bool {
+		guard let poolAddress = resource.metadata.poolUnit?.asGeneral else {
+			return false // no declared pool unit
+		}
+
+		guard ResourcePoolEntityType.addressSpace.contains(poolAddress.decodedKind) else {
+			return false // pool unit declared, but it is not a pool address. Invalid specification.
+		}
+
+		// Fetch pool unit info
+		let pool = try await getEntities(
+			[poolAddress],
+			.resourceMetadataKeys,
+			resource.atLedgerState,
+			.useCache
+		)
+		.map(\.resourcePool)
+		.first
+
+		guard let pool else {
+			return false // didn't load any pool
+		}
+
+		guard pool?.poolUnitResourceAddress == resource.resourceAddress else {
+			return false // The resource pool declared a different pool unit resource address
+		}
+
+		return true // It is a pool unit resource address
+	}
+
+	public func getPoolUnitDetails(_ poolUnitResource: OnLedgerEntity.Resource, forAmount amount: RETDecimal) async throws -> OwnedResourcePoolDetails? {
```

Comment:
Should this not be combined with `isPoolUnitResource`? It seems we will be loading the some of the same things just to figure out if it is a pool unit resource, we might as well go ahead and load the details? After making the sync checks locally of course. Or maybe there are situations where we only want the true/false answer?

Reply:
Btw, could you take the opportunity to rename `OwnedResourcePoolDetails`?